### PR TITLE
Add security support to LocalCluster.

### PR DESCRIPTION
**Pull request by mariusvniekerk**
This makes it easier for users to simulate a tls enabled cluster
locally.

### CODE REVIEW

File: distributed/deploy/tests/test_local.py [hunk 1]
```diff
@@ -444,5 +444,17 @@ def test_adapt_then_manual(loop):
...
+def test_local_tls(loop):
+    from distributed.utils_test import tls_only_security
+    security = tls_only_security()
+    with LocalCluster(scheduler_port=0, silence_logs=False, security=security,
+                      diagnostics_port=None, ip='tls://0.0.0.0', loop=loop) as c:
+        sync(loop, assert_can_connect_from_everywhere_4_6, c.scheduler.port,
+             protocol='tls',
+             connection_args=security.get_connection_args('client'),
+             timeout=1,
+             )
```
Comment (mrocklin) [inline distributed/deploy/tests/test_local.py]: Thoughts on adding some trivial assertions here like `assert c.scheduler.address.startswith('tls://')` or is the idea that the `assert_can_connect_from_everywhere` function should handle that for us?

**Comment by mrocklin** [comment]
In general this seems great to me.  I may not be well qualified to give technical review here though.

File: distributed/deploy/tests/test_local.py (hunk 1 above)
Comment (mariusvniekerk) [reply distributed/deploy/tests/test_local.py]: that was mostly my thought.  The other one that i should probably add is asserting that non-tls communication does not work,  and i guess more in the way of docs

**Comment by mrocklin** [comment]
This is causing a couple failures in tests.  They appear to be genuine.

**Comment by mariusvniekerk** [comment]
@mrocklin This seems to be passed now

**Comment by mariusvniekerk** [comment]
ah whoops, i was testing with docker :P

**Comment by mrocklin** [comment]
This is in.  My apologies for the late merge @mariusvniekerk , I was away last week. 

I also noticed that this is your first official code contribution to dask.distributed.  Welcome!

**Pull request merged**